### ISSUE TITLE
Data Corruption after power outage

### ISSUE THREAD

**Issue by sh0rez** [VictoriaMetrics/VictoriaMetrics#63]
Hi! After a datacenter crash related to a power outage, victoria is not able to come up anymore:

```
2019-06-11T14:49:39.922+0000    fatal   app/vmstorage/main.go:38
  cannot open a storage at victoria-metrics-data with retention period 120 months:
  cannot open table at "/victoria-metrics-data/data":
  cannot open partitions in the table "/victoria-metrics-data/data":
  cannot open partition "2019_06":
  cannot open small parts from "/victoria-metrics-data/data/small/2019_06":
  cannot open part "/victoria-metrics-data/data/small/2019_06/720_90_20190611054349.446_20190611054459.446_15A68A9DE06C8CA7": 
  cannot open timestamps file:
  open /victoria-metrics-data/data/small/2019_06/720_90_20190611054349.446_20190611054459.446_15A68A9DE06C8CA7/timestamps.bin:
  no such file or directory
```

Can I do anything about this?

**Comment by valyala** [comment]
It looks like VictoriaMetrics doesn't sync filenames inside created part directory before moving it from temporary path to the partition. This may result in disappeared files after power outage of system crash.
Will be fixed in the next release.

Temporary workaround: just remove the reported directory with missing file:
```
rm -rf /victoria-metrics-data/data/small/2019_06/720_90_20190611054349.446_20190611054459.446_15A68A9DE06C8CA7
```
This will result in the loss of 720 data points stored in this part.

**Comment by valyala** [comment]
The issue should be fixed in [v1.19.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by sh0rez** [comment]
Thank you for looking into this that quickly :)

**Comment by saurabh-hirani** [comment]
@valyala    We are on VM 1.58 and ran into the same issue 

```
victoria-metrics-20210408-064437-tags-v1.58.0-0-gedd1590ac
```

```
  panic    VictoriaMetrics/lib/fs/reader_at.go:92    FATAL: cannot open file "/storage/victoriametrics/vmdata/data/small/2021_05/9_9_20210523091719.393_20210523091719.393_1680BB8D781DF7C9/timestamps.bin" for reading: open /storage/victoriametrics/vmdata/data/small/2021_05/9_9_20210523091719.393_20210523091719.393_1680BB8D781DF7C9/timestamps.bin: no such file or directory
  ```

even though the said file is present. Happy to open a new issue and provide additional details. Had to delete these files which led to data loss but post that was able to access victoriametrics.

**Comment by saurabh-hirani** [comment]
Saw that the bug fix in v1.59 related to NFS as per https://docs.victoriametrics.com/CHANGELOG.html - deployed v1.60 - will update in case of any issues. Thanks.

**Comment by Wilderone** [comment]
face this issue on 1.72.0 cluster version, storage - AWS EFS 
```
{"ts":"2022-02-13T11:01:41.615Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:816","msg":"discarding /vm-data/cache/next_day_metric_ids, since it contains data for stale date; got 19012; want 19036"}
{"ts":"2022-02-13T11:01:41.619Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:258","msg":"opening table \"/vm-data/indexdb/16CFE4E29F51089B\"..."}
{"ts":"2022-02-13T11:01:41.689Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:293","msg":"table \"/vm-data/indexdb/16CFE4E29F51089B\" has been opened in 0.070 seconds; partsCount: 7; blocksCount: 250980, itemsCount: 54679304; sizeBytes: 2036301016"}
{"ts":"2022-02-13T11:01:41.690Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:258","msg":"opening table \"/vm-data/indexdb/16CB075C0867E1D3\"..."}
{"ts":"2022-02-13T11:01:41.770Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:293","msg":"table \"/vm-data/indexdb/16CB075C0867E1D3\" has been opened in 0.080 seconds; partsCount: 7; blocksCount: 420199, itemsCount: 90060277; sizeBytes: 3632049262"}
{"ts":"2022-02-13T11:01:41.813Z","level":"panic","caller":"VictoriaMetrics/lib/fs/reader_at.go:92","msg":"FATAL: cannot open file \"/vm-data/data/small/2022_01/49616118_280067_20220121111500.000_20220121154936.393_16CBEF605224DBDB/timestamps.bin\" for reading: open /vm-data/data/small/2022_01/49616118_280067_20220121111500.000_20220121154936.393_16CBEF605224DBDB/timestamps.bin: no such file or directory'}
```
